mods/ra/maps/soviet-04b/main.lua

```diff
@@ -0,0 +1,148 @@
+	end)
+
+	Trigger.OnEnteredFootprint(VillageCamArea, function(actor, id)
+		if actor.Owner == player then
+			Actor.Create("camera", true, { Owner = player, Location = VillageCamPoint.Location })
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ProduceInfantry)
+	Trigger.AfterDelay(DateTime.Minutes(2), ProduceArmor)
+
+	if Map.Difficulty == "Hard" or Map.Difficulty == "Medium" then
+		Trigger.AfterDelay(DateTime.Seconds(5), ReinfInf)
+	end
+	Trigger.AfterDelay(DateTime.Minutes(1), ReinfInf)
+	Trigger.AfterDelay(DateTime.Minutes(3), ReinfInf)
+	Trigger.AfterDelay(DateTime.Minutes(2), ReinfArmor)
+end
+
+Tick = function()
+	if Greece.HasNoRequiredUnits() then
+		player.MarkCompletedObjective(KillAll)
+		player.MarkCompletedObjective(KillRadar)
+	end
+
+	if player.HasNoRequiredUnits() then
+		Greece.MarkCompletedObjective(BeatUSSR)
+	end
+
+	if Greece.Resources >= Greece.ResourceCapacity * 0.75 then
+		Greece.Cash = Greece.Cash + Greece.Resources - Greece.ResourceCapacity * 0.25
+		Greece.Resources = Greece.ResourceCapacity * 0.25
+	end
+
+	if RCheck then
+		RCheck = false
+		if Map.Difficulty == "Hard" then
+			Trigger.AfterDelay(DateTime.Seconds(150), ReinfArmor)
+		elseif Map.Difficulty == "Medium" then
+			Trigger.AfterDelay(DateTime.Minutes(5), ReinfArmor)
+		else
+			Trigger.AfterDelay(DateTime.Minutes(8), ReinfArmor)
+		end
+	end
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("USSR")
+	Greece = Player.GetPlayer("Greece")
+
+	RunInitialActivities()
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	KillAll = player.AddPrimaryObjective("Defeat the Allied forces.")
+	BeatUSSR = Greece.AddPrimaryObjective("Defeat the Soviet forces.")
+	KillRadar = player.AddSecondaryObjective("Destroy Allied Radar Dome to stop enemy reinforcements.")
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "Lose")
+	end)
+
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "Win")
+	end)
+
+	Trigger.OnAnyKilled(Civs, function()
+		Trigger.ClearAll(civ1)
+		Trigger.ClearAll(civ2)
+		Trigger.ClearAll(civ3)
+		Trigger.ClearAll(civ4)
+		local units = Reinforcements.Reinforce(Greece, Avengers, { NRoadPoint.Location, ToVillageRoadPoint.Location }, 0)
+		Utils.Do(units, function(unit)
+			unit.Hunt()
+		end)
+	end)
+
+	Trigger.OnKilled(Radar, function()
+		player.MarkCompletedObjective(KillRadar)
+		Media.PlaySpeechNotification(player, "ObjectiveMet")
+	end)
+	
+	Trigger.OnDamaged(Harvester, function()
+		Guards.Guard(Harvester)
```

Comment:
This causes a crash.
The correct way would be
```lua
Utils.Do(Guards, function(unit)
	if not unit.IsDead and not Harvester.IsDead then
		unit.Guard(Harvester)
	end
end)
```
But anyway, I think `unit.AttackMove(Harvester.Location)` is better than guard, but that might be just me...